core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -2949,7 +2949,10 @@ struct ViewValueFunctor<DeviceType, ValueType, false /* is_scalar */> {
 
   template <typename Tag>
   void parallel_for_implementation() {
-    if (!space.in_parallel()) {
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
+    if (!space.in_parallel())
+#endif
+    {
```

Comment:
Wouldn't this deprecation need to come with a replacing if-condition so that even if `KOKKOS_ENABLE_DEPRECATED_CODE_4` is not defined, the behavior of this function remains unchanged?

Reply:
I think the non-`in_parallel` branch is never executed. Note that this function isn't even marked as `KOKKOS_FUNCTION`.

Reply:
Thus, I propose we just remove this branch in either case.